api.js

```diff
@@ -43,6 +43,22 @@ class Api extends EventEmitter {
 		const apiOptions = this.options;
 		runtimeOptions = runtimeOptions || {};
 
+		const {extensions: doNotCompileExtensions = []} = apiOptions;
+		const {extensions: babelExtensions = []} = apiOptions.babelConfig || {};
+
+		if (!apiOptions.babelConfig && doNotCompileExtensions.length === 0) {
+			doNotCompileExtensions.push('js');
+		}
+
+		if (apiOptions.babelConfig && babelExtensions.length === 0) {
+			babelExtensions.push('js');
+		}
+
+		// Combine all extensions possible for testing. Removing duplicate extensions.
+		const allExtensions = doNotCompileExtensions
+			.concat(babelExtensions)
+			.filter((ext, i, self) => self.indexOf(ext) === i);
```

Comment:
Would this be too much? `[...new Set([...doNotCompileExtensions, ...babelExtensions])]` 😊 

The `Set` constructor will deduplicate, and we can then turn it back into an array using the spread syntax.

Reply:
Oh i didn't know it would remove duplicates, that's pretty awesome actually (gotta change stuff in my personal projects).